pyproject.toml

```diff
@@ -31,25 +31,29 @@ exclude = '''
 '''
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = "^3.9"
 pymgclient = "^1.3.1"
 networkx = ">=2.5.1,<4.0.0"
 pydantic = "^2.3.0"
 psutil = ">=5.9,<7.0"
 dacite = "^1.6.0"
 adlfs = ">=2023.9.0,<2025.0.0"
 neo4j = ">=4.4.3,<6.0.0"
-numpy = "^1.24.1"
+numpy = "^1.26.4"
 
-pyarrow = { version = "^13.0.0", optional = true }
-torch = { version = "^1.13.1", optional = true }
-dgl = { version = "^0.9.1", optional = true }
+pyarrow = { version = "^18.0.0", optional = true }
+torch = { version = "^2.4", optional = true, source = "pytorch-cpu" }
 docker = { version = "^6.1.3", optional = true }
 
+[[tool.poetry.source]]
+name = "pytorch-cpu"
+url  = "https://download.pytorch.org/whl/cpu"
+priority = "explicit"
+
 [tool.poetry.extras]
 arrow = ["pyarrow"]
 dgl = ["torch", "dgl"]
-all = ["pyarrow", "torch", "dgl", "docker"]
+all = ["pyarrow", "torch", "docker"]
```

Comment:
This means that in the installation process, users won't be able to use all GQLAlchemy capabilities when installing with all extras? How does this affect the installation instructions from README, both on downloading from pypi and building from source?